pytrip/cube.py

```diff
@@ -730,6 +730,7 @@ def read(self, path):
 
         :param str path: Path to filename to be read, file extention may be given but is not neccesary.
         """
+        self.basename =  os.path.basename(path).split(".")[0]
```

Comment:
I see the point. There is already a place where path parsing is done, now it is in `read_trip_data_file`:

```
        # extract header and data file name from path
        header_file_name, data_file_name = self.parse_path(path)
```

I'd propose to move setting basename there (from `read` into `read_trip_data_file`) to keep all path related code in single place.  
Then in the future it could become an element of some more sophisticated path parsing engine.